services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/extension/env/EnvCleanupExtension.groovy

```diff
@@ -0,0 +1,155 @@
+
+import groovy.util.logging.Slf4j
+import org.spockframework.runtime.extension.AbstractGlobalExtension
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+
+@Slf4j
+abstract class EnvCleanupExtension extends AbstractGlobalExtension implements SpringContextListener {
+
+    @Autowired
+    TopologyDefinition topology
+
+    @Autowired
+    NorthboundService northbound
+
+    @Autowired
+    Database database
+
+    @Autowired
+    IslUtils islUtils
+
+    @Autowired
+    LockKeeperService lockKeeper
+
+    @Value('${spring.profiles.active}')
+    String profile
+
+    @Override
+    void start() {
+        SpringContextExtension.addListener(this)
+    }
+
+    def deleteAllFlows() {
+        log.info("Deleting all flows")
+        northbound.deleteAllFlows()
+    }
+
+    def unsetIslMaintenance(List<IslInfoData> links) {
+        def maintenanceLinks = northbound.getAllLinks().findAll { it.underMaintenance }
+        if (maintenanceLinks) {
+            log.info("Unset maintenance mode for links: $maintenanceLinks")
+            maintenanceLinks.each {
+                northbound.setLinkMaintenance(islUtils.getLinkUnderMaintenance(it, false, false))
+            }
+        }
+    }
+
+    def reviveFailedLinks(List<IslInfoData> links) {
+        def failedLinks = links.find { it.state == IslChangeType.FAILED }
+        if (failedLinks) {
+            log.info("Bring ports up for failed links: $failedLinks")
+            failedLinks.each {
+                try {
+                    northbound.portUp(it.source.switchId, it.source.portNo)
+                    northbound.portUp(it.destination.switchId, it.destination.portNo)
+                } catch (Throwable t) {
+                    // Switch may be disconnected. Ignore for now, healthcheck test will report problems if any.
+                    // This situation may require manual intervention
+                }
+            }
+        }
+    }
+
+    def deleteLinkProps() {
+        def linkProps = northbound.getAllLinkProps()
+        if (linkProps) {
+            log.info("Deleting all link props")
+            northbound.deleteLinkProps(linkProps)
+        }
+    }
+
+    def resetCosts() {
+        log.info("Resetting all link costs")
+        database.resetCosts()
+    }
+
+    def resetBandwidth(List<IslInfoData> links) {
+        def topoIsls = topology.isls.collectMany { [it, islUtils.reverseIsl(it)] }
+        links.each { link ->
+            if (link.maxBandwidth != link.availableBandwidth || link.maxBandwidth != link.speed) {
+                def isl = topoIsls.find {
+                    it.srcSwitch.dpId == link.source.switchId && it.srcPort == link.source.portNo &&
+                            it.dstSwitch.dpId == link.destination.switchId && it.dstPort == link.destination.portNo
+                }
+                assert isl
+                log.info("Resetting available bandwidth on ISL $isl")
```

Comment:
The same question with printing as above.

Reply:
This is an absolutely abnormal and rare situation, which can moreover mask a defective ISL discovery by Kilda. From my point of view it even deserves to be of a 'warn' level.

Reply:
Ok. Please add ':' after 'ISL' to be consistent with other messages.

Reply:
done